From: Juergen Borleis <jbe@pengutronix.de>
Date: Tue, 1 Dec 2015 12:31:31 +0100
Subject: [PATCH] DTS/Skov-GPIO: Re-factor the pin setup for the board's
 version/variant encoding

These pins are never changing input only pins.

Signed-off-by: Juergen Borleis <jbe@pengutronix.de>
---
 arch/arm/boot/dts/imx6qdl-cpu.dtsi | 24 ++++++++++++++++--------
 1 file changed, 16 insertions(+), 8 deletions(-)

diff --git a/arch/arm/boot/dts/imx6qdl-cpu.dtsi b/arch/arm/boot/dts/imx6qdl-cpu.dtsi
index 71520295bc29..3843c685a797 100644
--- a/arch/arm/boot/dts/imx6qdl-cpu.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-cpu.dtsi
@@ -57,6 +57,8 @@
 	};
 
 	skov_version {
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_skov_version>;
 		compatible = "skov-version";
 		v0-gpio = <&gpio2 10 0>;
 		v1-gpio = <&gpio2 15 0>;
@@ -340,14 +342,20 @@
 				MX6QDL_PAD_GPIO_5__GPIO1_IO05		0xb0		/* ethernet phy reset */
 				MX6QDL_PAD_EIM_D30__GPIO3_IO30		0xb0		/* ethernet phy interrupt */
 				MX6QDL_PAD_ENET_RX_ER__GPIO1_IO24	0x13000		/* Enet */
-				MX6QDL_PAD_SD4_DAT0__GPIO2_IO08		0x1b0b0		/* VC0 */
-				MX6QDL_PAD_SD4_DAT1__GPIO2_IO09		0x1b0b0		/* VC1 */
-				MX6QDL_PAD_SD4_DAT2__GPIO2_IO10		0x1b0b0		/* VC2 */
-				MX6QDL_PAD_SD4_DAT3__GPIO2_IO11		0x1b0b0		/* VC3 */
-				MX6QDL_PAD_SD4_DAT4__GPIO2_IO12		0x1b0b0		/* VC4 */
-				MX6QDL_PAD_SD4_DAT5__GPIO2_IO13		0x1b0b0		/* VC5 */
-				MX6QDL_PAD_SD4_DAT6__GPIO2_IO14		0x1b0b0		/* VC6 */
-				MX6QDL_PAD_SD4_DAT7__GPIO2_IO15		0x1b0b0		/* VC7 */
+			>;
+		};
+
+		pinctrl_skov_version: skov_version_pins {
+			fsl,pins = <
+				/* never changing input pins */
+				MX6QDL_PAD_SD4_DAT0__GPIO2_IO08		0x1b040		/* VC0 */
+				MX6QDL_PAD_SD4_DAT1__GPIO2_IO09		0x1b040		/* VC1 */
+				MX6QDL_PAD_SD4_DAT2__GPIO2_IO10		0x1b040		/* VC2 */
+				MX6QDL_PAD_SD4_DAT3__GPIO2_IO11		0x1b040		/* VC3 */
+				MX6QDL_PAD_SD4_DAT4__GPIO2_IO12		0x1b040		/* VC4 */
+				MX6QDL_PAD_SD4_DAT5__GPIO2_IO13		0x1b040		/* VC5 */
+				MX6QDL_PAD_SD4_DAT6__GPIO2_IO14		0x1b040		/* VC6 */
+				MX6QDL_PAD_SD4_DAT7__GPIO2_IO15		0x1b040		/* VC7 */
 			>;
 		};
 
